crates/install-wheel-rs/src/linker.rs

```diff
@@ -302,22 +302,37 @@ fn clone_recursive(site_packages: &Path, wheel: &Path, entry: &DirEntry) -> Resu
     debug!("Cloning {} to {}", from.display(), to.display());
 
     // Attempt to copy the file or directory
-    let reflink = reflink_copy::reflink(&from, &to);
+    let mut reflink = reflink_copy::reflink_or_copy(&from, &to);
```

Comment:
Can we try this without using `reflink_or_copy`? The workaround here feels like it's not quite the right tradeoff. `reflink_or_copy` just calls reflink and then, if it fails, it calls copy. Can we just mimic that here? We should be able to tell from the error directly without having to do the additional `reflink` call.

The other problem with the current approach is that we're going to try and `reflink` for every file, even though we _know_ it won't work. (If the first reflink fails for any reason other than `AlreadyExists`, then all of the subsequent reflinks will fail too.) In `hardlink_wheel_files`, we avoid this by _just_ copying once we know we can't hardlink.

For example, can you try something like: still using `reflink_copy::reflink` here, then adding another branch below, where if the error is anything other than `AlreadyExists`, we fall back to recursively copying the directory.

Reply:
You're right, this could probably be a lot simpler. Technically, `reflink_copy::reflink_or_copy` works a bit differently than separate `reflink_copy::reflink` + `std::fs::copy` calls, but it's not that important of a difference.

And yes, we should definitely try to avoid doing extra work here.